roles/postgresql/tasks/main.yml

```diff
@@ -1,4 +1,11 @@
 ---
+- name: Remove pre-installed PostgreSQL versions
+  become: true
+  apt:
+    name: postgresql*
```

Comment:
I'm wondering about possible side effects when the installer is executed again on existing installations (some people do so) :thinking:. Even if we don't officially support it, making the installer idempotent (or as idempotent as possible) would be nice.

So how about:

```
- name: Remove unsupported PostgreSQL versions
  become: true
  apt:
    name:
      - postgresql*9.3*
      - postgresql*9.2*
      - postgresql*9.1*
      - postgresql*9.0*
    state: absent
```

Reply:
Although it looks like this code isn't working on Ubuntu 20.04 builds :thinking:. On Ubuntu 16.04 and 18.04 builds it works even though versions 9.0 through 9.2 are not installed there.

Reply:
Glad you realized about the idempotency issue ☺️:

I needed to add an extra line to your ansible task as it fails when there is no package with the given name. The error I get is the following:

```
TASK [postgresql : Remove unsupported PostgreSQL versions] **************************************************************************************************************************************************************************************************************
fatal: [192.168.33.184]: FAILED! => changed=false 
  msg: No package(s) matching 'postgresql*9.3*' available
```
I think this is related to this Ansible issue which seems to be still opened:

https://github.com/ansible/ansible/issues/62262

As this command fails when there is no package matching the given wildcard but works otherwise, maybe we can ignore this task's errors.

```
- name: Remove unsupported PostgreSQL versions
   become: true 
   apt: 
     name: 
       - postgresql*9.3* 
       - postgresql*9.2* 
       - postgresql*9.1* 
       - postgresql*9.0* 
      state: absent
      ignore_errors: true
```

A more complex workaround could be something like this:

```
- name: Remove unsupported PostgreSQL versions
  become: true
  block:
    - name: "Check if PostreSQL 9.3 is installed"
      package_facts:
        manager: "auto"
    - name: "Remove PostreSQL 9.3"
      apt:
        name:
          - postgresql*9.3*
        state: absent
      when: "'postgresql-9.3' in ansible_facts.packages"
    - name: "Remove PostreSQL 9.2"
      apt:
        name:
          - postgresql*9.2*
        state: absent
      when: "'postgresql-9.2' in ansible_facts.packages"
    - name: "Remove PostreSQL 9.1"
      apt:
        name:
          - postgresql*9.1*
        state: absent
      when: "'postgresql-9.1' in ansible_facts.packages"
    - name: "Remove PostreSQL 9.0"
      apt:
        name:
          - postgresql*9.0*
        state: absent
      when: "'postgresql-9.0' in ansible_facts.packages"
```